lib/tasks/rh_cloud_inventory.rake

```diff
@@ -27,6 +42,14 @@ namespace :rh_cloud_inventory do
         end
       end
     end
+    desc 'Upload generated inventory report to Red Hat cloud'
+    task upload: [:environment, 'dynflow:client'] do
```

Comment:
I see `dynflow:client` was not enough. Got exception thrown at me:
```
2020-11-17T18:10:44 [I|app|] Performing ForemanInventoryUpload::Async::QueueForUploadJob (Job ID: c7953845-ebe4-415d-b79e-df803b51d5f1) from Dynflow(default) enqueued at  with arguments: "/home/vagrant/foreman/red_hat_inventory/generated_reports/", "report_for_1.tar.gz", "1"
2020-11-17T18:10:44 [D|bac|] Ensuring objects
2020-11-17T18:10:44 [D|bac|] Copying report_for_1.tar.gz to /home/vagrant/foreman/red_hat_inventory/uploads/
2020-11-17T18:10:44 [D|bac|] Done copying report_for_1.tar.gz to /home/vagrant/foreman/red_hat_inventory/uploads/report_for_1.tar.gz
E, [2020-11-17T18:10:44.761724 #30440] ERROR -- /connector-database-core: No executor available (Dynflow::Error)
/home/vagrant/foreman/.vendor/ruby/2.5.0/gems/dynflow-1.4.7/lib/dynflow/connectors/database.rb:143:in `any_executor'
/home/vagrant/foreman/.vendor/ruby/2.5.0/gems/dynflow-1.4.7/lib/dynflow/connectors/database.rb:135:in `find_receiver'
...
/home/vagrant/foreman/.vendor/ruby/2.5.0/gems/activejob-6.0.3.4/lib/active_job/enqueuing.rb:22:in `perform_later'
/home/vagrant/foreman_rh_cloud/lib/foreman_inventory_upload/async/queue_for_upload_job.rb:15:in `perform'
```

@adamruzicka, maybe you know a better solution to this?

Reply:
Hm, I didn't get any exception on Satellite 6.8.1 (Foreman 2.1.2.20) or 6.9.0 (Foreman 2.3.0).

Reply:
Depending on `dynflow:client` marks the task as "allowed to communicate with dynflow, without having the capability to execute jobs on its own", which is fine. Since we're using dynflow as active job backend, submitting activejob jobs using `klass.perform_later` will require dynflow executor (dynflowd or dynflow-sidekiq@orchestrator) running. Without it it cannot perform those actions in background.

We could make the rake task act as an executor, but setting that up would be a bit more involved.